src/main/java/software/amazon/encryption/s3/materials/MaterialsDescription.java

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.amazon.encryption.s3.materials;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+/**
+ * This class is used to store and manage key-value pairs that describe keyring,specifically for AES and RSA Keyring.
+ * This will be useful during re-encryption of instruction file.
+ * The stored Materials Description are immutable once created.
+ */
+
+public class MaterialsDescription {
```

Comment:
We can have this _be_ a `Map<String, String>` extension, meaning: 

```
implements Map<String, String>
```

We'll need to implement a bunch of other methods, but they can all be deferred to the "inner" map pretty easily. This is a nicer customer experience as customers can use all of the usual `Map` API and aren't limited to the things we chose to implement.